playbooks/production/roles/common/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+- name: Update apt cache
+  apt: update_cache=yes cache_valid_time=3600
+
+- name: Install packages
+  apt: name={{ item }}
+  with_items:
+    - git
+    - docker.io
+    - docker-compose
+    - python3-pip
+    - python3-venv
+    - curl
+    - libssl-dev
+    - pkg-config
+
+#- name: Add to fstab
+#  ansible.builtin.lineinfile:
+#    path: /etc/fstab
+#    line: "UUID=64d72737-f1de-4f67-a22a-f693a79f228f /data ext4 defaults,nofail 0 2"
+#    state: present
+
+- name: Mount the disk
+  ansible.posix.mount:
+    path=/data src=UUID=64d72737-f1de-4f67-a22a-f693a79f228f fstype=ext4 state=mounted
+  # its ok if it fails (for simulation purposes)
+  ignore_errors: true
+
+
+- name: Create data directories if they don't exist
+  ansible.builtin.file: path={{ item.data_dir }} state=directory
+  with_items: "{{ vhosts }}"
+
+- name: Create data directory for storage
+  ansible.builtin.file: path=/data/storage state=directory
+
+- name: Verify if Rust installed
+  shell: rustc --version
+  register: rust_installed
+  ignore_errors: true
+
+- name: Install Rust
+  when: rust_installed.rc != 0
+  shell: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+  args:
+    creates: /root/rustup-init
+
+- name: Is cargo in PATH?
+  shell: echo $PATH | grep -q "$HOME/.cargo/bin"
+  register: cargo_in_path
+  ignore_errors: true
+
+- name: Add cargo to PATH
+  when: cargo_in_path.rc != 0
+  shell: echo 'export PATH="$HOME/.cargo/bin:$PATH"' >> /root/.bashrc
+  args:
+    creates: /root/.bashrc
+
+- name: Delete old kernelci-storage directory
+  file: path=/root/kernelci-storage state=absent
+  ignore_errors: true
+
+- name: Clone storage repo
+  git:
+    repo: https://github.com/nuclearcat/kernelci-storage.git
+    dest: /root/kernelci-storage
+    update: true
+    force: true
+
+- name: Build Rust program kernelci-storage
+  shell: cd /root/kernelci-storage && cargo build --release
+  args:
+    creates: /root/kernelci-storage/target/release/kernelci-storage
+
+- name: Install kernelci-storage to /usr/local/bin
+  shell: mv /root/kernelci-storage/target/release/kernelci-storage /usr/local/bin
```

Comment:
I'd suggest setting up kernelci-storage releases externally and just retrieving the binary to the servers.

This way we limit software deployed on the server to reduce potential attacks surface and overall clutter in the systems ;)

Reply:
It's great to have `kernelci-storage` now [packaged into Docker container image](https://github.com/nuclearcat/kernelci-storage/pkgs/container/kernelci-storage), it should be a good fit as a replacement for these tasks